### PR TITLE
skip writing content when status code is 204 (StatusNoContent)

### PR DESCRIPTION
**Pull request by amit-lamba**

### CODE REVIEW

**Review by argouber — COMMENTED** [review]
Could you add a unit-test for the 204.
Also, are there other responses that should not have body?

**Comment by amit-lamba** [comment]
> Could you add a unit-test for the 204.
> Also, are there other responses that should not have body?

Added the unit-test.
Through the logs, only the one which have a 204 were having this error log.
In other cases even if the body is blank should not cause this issue.

**Review by argouber — APPROVED** [review]

**Comment by rpatali** [comment]
> Could you add a unit-test for the 204.
> Also, are there other responses that should not have body?

304

File: runtime/server_http_response.go [hunk 1]
```diff
@@ -291,7 +292,9 @@ func (res *ServerHTTPResponse) flush(ctx context.Context) {
 
 	res.flushed = true
 	res.writeHeader(res.pendingStatusCode)
-	res.writeBytes(res.pendingBodyBytes)
+	if res.pendingStatusCode != http.StatusNoContent {
```
Comment (rpatali) [inline runtime/server_http_response.go:295]: we use 304 for use cases so you probably want to accommodate for that status code as well.

File: runtime/constants.go [hunk 2]
```diff
@@ -137,3 +137,8 @@ var knownStatusCodes = map[int]bool{
 	http.StatusNotExtended:                   true, // 510
 	http.StatusNetworkAuthenticationRequired: true, // 511
 }
+
+var noContentStatusCodes = map[int]bool{
```
Comment (rpatali) [inline runtime/constants.go:141]: noBodyStatusCodes or something similar would be better since `noContentStatusCode` implies only 204.

**Review by rpatali — APPROVED** [review]

File: runtime/constants.go (hunk 2 above)
Comment (amit-lamba) [reply runtime/constants.go:141]: I think, Body and Content sound similar, and noBodyStatusCode**s** plural may signify that we are including all no content ones, and avoid confusion.

**Pull request merged**